lib/helper/TokenHolder.js

```diff
@@ -0,0 +1,149 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+'use strict';
+
+const TxSender = require('./../../utils/TxSender'),
+  AbiBinProvider = require('./../AbiBinProvider');
+
+const tokenHolderContractName = 'TokenHolder',
+  tokenRulesContractName = 'TokenRules';
+
+/**
+ * It consists of methods for a user's TokenHolder.
+ */
+class TokenHolder {
+  /**
+   * Constructor of TokenHolder.
+   *
+   * @param auxiliaryWeb3 Auxiliary web3 object.
+   * @param tokenRules Token rules contract address.
+   * @param tokenHolderProxy TokenHolder proxy address of a user.
+   */
+  constructor(auxiliaryWeb3, tokenRules, tokenHolderProxy) {
+    const oThis = this;
+
+    oThis.auxiliaryWeb3 = auxiliaryWeb3;
+    oThis.tokenRules = tokenRules;
+    oThis.tokenHolderProxy = tokenHolderProxy;
+    oThis.abiBinProvider = new AbiBinProvider();
+  }
+
+  /**
+   * It is used to get the executable data for directTransfers method of TokenRules.
+   *
+   * @param transferTo List of addresses to transfer.
+   * @param transfersAmount List of amounts to transfer.
+   *
+   * @returns Executable data of directTransfers method.
+   */
+  getDirectTransferExecutableData(transferTo, transfersAmount) {
+    const oThis = this;
+
+    const jsonInterface = oThis.abiBinProvider.getABI(tokenRulesContractName),
+      tokenRuleContract = new oThis.auxiliaryWeb3.eth.Contract(jsonInterface, oThis.tokenRules),
+      directTransferExecutable = tokenRuleContract.methods.directTransfers(transferTo, transfersAmount).encodeABI();
+
+    return directTransferExecutable;
+  }
+
+  /**
+   * It is used to get call prefix of executeRule method in TokenHolder contract.
+   *
+   * @returns Encoded signature of executeRule method.
+   */
+  getTokenHolderExecuteRuleCallPrefix() {
+    const oThis = this;
```

Comment:
I will suggest to query it from contract ABI. There is already a constant in TH contract: EXECUTE_RULE_CALLPREFIX